### PR TITLE
[CARBONDATA-2296] Fix create datamap command with out on table syntax and correct the target location of test framework

### PR DESCRIPTION
**Pull request by ravipesala**
Problem 
1 Create datamap command fails if user does not mention `on table` 
2 Test framework try to create metastore and store location in integration/spark-common location but if it runs from other modules like datamap it creates target location in wrong place.
Solution
Check the null for parent table if not present and always take the present module target location as store path location if it not under integration module.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4699/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3472/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4207/

File: core/src/main/java/org/apache/carbondata/core/datamap/IndexDataMapProvider.java [hunk 1]
```diff
@@ -41,6 +41,9 @@ public IndexDataMapProvider(DataMapSchemaStorageProvider storageProvider) {
   @Override
   public void initMeta(CarbonTable mainTable, DataMapSchema dataMapSchema, String ctasSqlStatement)
       throws MalformedDataMapCommandException, IOException {
+    if (mainTable == null) {
+      throw new MalformedDataMapCommandException("Parent table required to create index");
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datamap/IndexDataMapProvider.java]: change to `Parent table is required to create datamap`

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/datamap/CarbonCreateDataMapCommand.scala [hunk 2]
```diff
@@ -64,7 +64,7 @@ case class CarbonCreateDataMapCommand(
     }
 
     dataMapSchema = new DataMapSchema(dataMapName, dmClassName)
-    if (mainTable.isStreamingTable &&
+    if (mainTable != null && mainTable.isStreamingTable &&
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/datamap/CarbonCreateDataMapCommand.scala]: put ` mainTable.isStreamingTable` to next line

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4701/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3474/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4209/

**Comment by jackylk** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3475/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4210/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4702/

**Pull request closed without merging**